### DISCUSSION TITLE
Minimal KDE alternative profile

### DISCUSSION THREAD

**Discussion by psyfry** [archlinux/archinstall#750] [Ideas]
Floating around an idea based on a common support question that gets asked frequently on Reddit and other Arch communities regarding the KDE profile. Basically, what happens is a user wants KDE, they install with archinstall, get 'full kde'(games, redundant applications, the _whole_ KDE suite), and then run into difficulties trying to remove specific packages from the plasma-meta package. The issue is the classic "minimal plasma" vs. "full kde" situation that happens on every distro, where starting with full kde from a meta/group package and subtracting unwanted programs is substantially more difficult than starting with plasma-desktop and adding non-meta/group programs that are wanted. The advice is usually along the lines of "reinstall without the meta package while your ahead".

I did a small experiment with replacing **plasma-meta** in the kde profile with **plasma-desktop** and the result is a working, serviceable minimal plasma installation. I think there's some significant utility for new arch users having the option of minimal vs full kde, since I think many of the new users using archinstall have the expectation/goal of the "minimal arch with plasma" starting place that they can build upon and customize rather than the convenient fully-featured kde install they could get with any other distro. 

If there's any interest, I'd be glad to work out all the details and put together a PR.

**Comment by Torxed** [comment]
We've been discussing this a few times, either with GNOME or with KDE. The end result is usually that we cannot and should not currate and maintain a list of individual packages. We therefore rely on the upstream group or meta packages.

If there's a "minimal" (but working user experience) package group, I guess plasma-dekstop then I think we should be using that instead of meta.

Historically speaking that has caused too many applications to not function as expected (app stores, backgrounds, settings, graphical editor and so on) so we've had to use the meta package to not get flooded with "broken KDE".

If there's no objections, and you have spare time I'd welcome a simple PR that swaps the meta package group. That way we don't need to have another option between full and minimal, users tend to opt in for full by default and then we're back to square one hehe.

Edit: Forgot to say, thank you for supporting the project by answering support questions in the different forums!

**Comment by Col433** [comment]
@psyfry I installed plasma-desktop plasma-wayland-session konsole sddm dolphin to get a minimal working KDE install and had no problems but you do need to add those extra packages to plasma-desktop. I was just interested to see if using a wayland session at the moment I had any better performance but for my configuration there was none still going forward wayland is probably the future for most distros. I use xfce4 minimal for my install at add extra packages and skip the DE options. While it's great to have some popular DE options there  maintaining archinstall with a plethora of them to me is looking for unnecessary trouble. I would also be interested to see you develop a minimal KDE PR though especially if you also include the wayland session option. Looks like we will be waiting some time before xfce supports a wayland session...

**Comment by djwyman** [comment]
Is there a particular reason that plasma-meta is installed vs plasma group?  I recently used "archinstall" on my laptop and when I tried to remove discover (why is discover even default in an arch  meta/group to begin with?) but when I did, I found that it was a dependency of plasma-meta.  Removing plasma-meta then orphans the rest of plasma, whereas if plasma group was used, discover (or some other part) could be removed without having to mark all the other packages to explicit manually.

**Reply by potamoi** [reply]
My understanding is that meta packages automatically install new packages when the meta package itself is updated with new dependencies. This is done to prevent installations from breaking when a new dependency is needed.

Groups do not do this, so installing via this method can end up breaking things within KDE or KDE itself when packages get updated, but the new dependencies required do not get installed.

The only real way to fix this would be to try and get the Arch maintainers of the **plasma-meta** package to remove unnecessary packages (i.e discover), or for **archinstall** to curate its own list, but this just creates more maintenance.

I suppose the only other option is to use the **plasma-desktop** package, but this can create its own issues with things not working as the user expected, as @torxed has mentioned above.

Personally, I would go for the **plasma-desktop** method – as long as the user has a bootable DE, it should be trivial for them to make any tweaks or install any other packages they require.
